packages/vite/src/node/__tests__/plugins/importGlob/fixture-a/index.ts

```diff
@@ -67,3 +67,13 @@ export const cleverCwd2 = import.meta.glob([
   '../fixture-b/*.ts',
   '!**/index.ts',
 ])
+
+export const customBase = import.meta.glob('./**/*.ts', { base: './' })
+
+export const customRootBase = import.meta.glob('./**/*.ts', {
+  base: '/fixture-b',
+})
+
+export const aliasBase = import.meta.glob('@modules/*.ts', {
+  base: '/fixture-a/modules',
+})
```

Comment:
To test this correctly, this `resolveId` function needs to have an alias feature supported.
https://github.com/vitejs/vite/blob/500cb145d9f3bc947387bb0c286cd0f427effc9f/packages/vite/src/node/__tests__/plugins/importGlob/fixture.spec.ts#L11
```ts
const resolveId = (id) => {
  if (id.startsWith('@modules/')) {
    return normalizePath(path.resolve(root, 'modules', id.slice('@modules/'.length)))
  }
  return id
}
```

I expect the output to be same with
```js
import.meta.glob('@modules/*.ts')
```
if the alias works like above.

I think we should also test:
```ts
import.meta.glob('./*.ts', {
  base: '@modules',
})
```

Reply:
> I expect the output to be same with
> 
> ```js
> import.meta.glob('@modules/*.ts')
> ```

You're right. I was actually suggesting a different output before (https://github.com/vitejs/vite/pull/18510#discussion_r1827147602). Since `base` is now where we resolve the globs from, but aliases will always return a root-relative key, that might be one of the downsides if treating `base` this way I think. Unless we want to make `base` unique where, if it's specified, we'll resolve the keys relative to it?

> I think we should also test:
> 
> ```ts
> import.meta.glob('./*.ts', {
>   base: '@modules',
> })
> ```

The implementation I had in mind when I added my reviews before is that `base: '@modules'` is a little tricky to support, and we can only support those that starts with `/` or `./` or `../`. Since we're trying to figure out the `dir` to resolve from, and resolving alias would have to involve `resolveId`, but I don't think it handles resolving dirs well, only files.

Reply:
> Since we're trying to figure out the `dir` to resolve from, and resolving alias would have to involve `resolveId`, but I don't think it handles resolving dirs well, only files.

We are passing `glob` to `resolveId` here.
https://github.com/vitejs/vite/blob/500cb145d9f3bc947387bb0c286cd0f427effc9f/packages/vite/src/node/plugins/importMetaGlob.ts#L594-L598
Maybe the dirs can be resoved by something like `resolveId(base + '*').slice(0, -'*'.length)`?

Reply:
Hmm yeah I guess if globs like `@alias/*.ts` can already be resolved here, then `base` should be able to do the same here too. Though if it's harder to get it right I was thinking it could be implemented later in the future. But fine by me either ways.